### ISSUE TITLE
[Bug]: [partitionkey] should report error if num_partitions=0

### ISSUE THREAD

**Issue by yanliang567** [milvus-io/milvus#24140]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230515-8b8f9f1e
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):   pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus-2.2.9.dev8
```

### Current Behavior

create collection successfully with 64(default) partitions, when setting num_partitions=0

### Expected Behavior

milvus report a meaningful error msg

### Steps To Reproduce

```markdown
1. create a collection schema with partition key on
2. create a collection with num_partitions=0
```

**Comment by xige-16** [comment]
fixed on pymilvus

**Comment by yanliang567** [comment]
> also when setting
> 
>     1. num_partitions=None, it creates 64 partitions
> 
>     2. num_partitions=True, it creates 1 partition
> 
>     3. num_partitions=False, it creates 64 parititions
>        if possible, I prefer reporting error msgs for all above

#1 is expected in pymilvus after discussion. #2 and #3 verified fix on pymilvus-2.2.9.dev11
